### PR TITLE
Add reference page for required permissions to install KIC

### PR DESCRIPTION
**Pull request by randmonkey**
### Description

Add a reference page to list the permissions required to install KIC when user do not have the super admin permission.
Fixes https://github.com/Kong/kubernetes-ingress-controller/issues/5166. 
 

### Testing instructions

Preview link: 
https://deploy-preview-6524--kongdocs.netlify.app/kubernetes-ingress-controller/latest/reference/required-permissions/
### Checklist 

- [x] Review label added 
- [ ] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_src/kubernetes-ingress-controller/reference/required-permissions.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+title: Required Permissions to Use {{ site.kic_product_name }}
+type: reference
+purpose: |
+  What permissions are required to install {{ site.kic_product_name }} if I am not a super admin of the cluster? What perssions are required to run {{ site.kic_product_name }}?
+---
+
+To install {{ site.kic_product_name }}, you need to have the perssion to WRITE (create, update, get, list, watch in Kubernetes's RBAC model) the following resources:
```
Comment (tao12345666333) [inline app/_src/kubernetes-ingress-controller/reference/required-permissions.md:8]:
```suggestion
To install {{ site.kic_product_name }}, you need to have the permission to WRITE (create, update, get, list, watch in Kubernetes's RBAC model) the following resources:
```

File: app/_src/kubernetes-ingress-controller/reference/required-permissions.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+For creating RBAC rules to enable {{ site.kic_product_name }} to access requried resources, you need to have the permissions to create or update `ClusterRole`, `ClusterRoleBinding` in the cluster scope, and create or update `Role`,`RoleBinding` in the  namespace in which you want to install {{ site.kic_product_name }}.
```
[the remark this answers is not in the document]
Comment (randmonkey) [reply app/_src/kubernetes-ingress-controller/reference/required-permissions.md]: done.

File: app/_src/kubernetes-ingress-controller/reference/required-permissions.md (hunk 1 above)
Comment (randmonkey) [reply app/_src/kubernetes-ingress-controller/reference/required-permissions.md]: done.

**Comment by randmonkey** [comment]
I added `if ${{ failure() }}` condition, but logs are still uploaded. hold this.

File: app/_src/kubernetes-ingress-controller/reference/required-permissions.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+---
+title: Required Permissions to Use {{ site.kic_product_name }}
+type: reference
+purpose: |
+  What permissions are required to install {{ site.kic_product_name }} if I am not a super admin of the cluster? What permissions are required to run {{ site.kic_product_name }}?
+---
+
+To install {{ site.kic_product_name }}, you need to have the permissions to WRITE (create, update, get, list, watch in Kubernetes's RBAC model) the following resources:
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/reference/required-permissions.md:8]:
```suggestion
To install {{ site.kic_product_name }}, you need to have the permissions to WRITE (create, update, get, list, watch in Kubernetes's RBAC model) these resources.
```

**Review by Rajakavitha1 — APPROVED** [review]
LGTM

File: app/_src/kubernetes-ingress-controller/reference/required-permissions.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+---
+title: Required Permissions to Use {{ site.kic_product_name }}
```
Comment (lena-larionova) [inline app/_src/kubernetes-ingress-controller/reference/required-permissions.md:2]:
```suggestion
title: Required permissions to use Kong Ingress Controller
```
variables don't work right in front matter, unfortunately:
Screenshot 2023-11-30 at 11 49 04 AM

File: app/_src/kubernetes-ingress-controller/reference/required-permissions.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+type: reference
+purpose: |
+  What permissions are required to install {{ site.kic_product_name }} if I am not a super admin of the cluster? What permissions are required to run {{ site.kic_product_name }}?
```
Comment (lena-larionova) [inline app/_src/kubernetes-ingress-controller/reference/required-permissions.md:5]:
```suggestion
  What permissions are required to install Kong Ingress Controller if I am not a super admin of the cluster? What permissions are required to run Kong Ingress Controller?
```

File: app/_src/kubernetes-ingress-controller/reference/required-permissions.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+For creating RBAC rules to enable {{ site.kic_product_name }} to access requried resources, you need to have the permissions to create or update `ClusterRole`, `ClusterRoleBinding` in the cluster scope, and create or update `Role`,`RoleBinding` in the namespace in which you want to install {{ site.kic_product_name }}.
+
+To install CRDs required to configure Kong specific entities, like `KongPlugin` to configure plugins, you need to have the permissions to create and update `CustomResourceDefinition`.
+
+For creating resources required for {{ site.kic_product_name }}, you need to have the permissions to create or update `Deployment`, `Service`, `ServiceAccount`, `Secret` or `ConfigMap` to create deployments, services and specify their configurations. If you run {{ site.kic_product_name }} with database backed {{site.base_gateway}}, you also need to have the permissions to create or update `PersistentVolumeClaim` to set volumes used for database, and `Job` to run migration jobs. If you want to enable auto scaling, permissions to create or update `HorizontalPodAutoscaler`.
```
Comment (lena-larionova) [inline app/_src/kubernetes-ingress-controller/reference/required-permissions.md:14]:
```suggestion
* For creating resources required for {{ site.kic_product_name }}, you need to have the permissions to create or update `Deployment`, `Service`, `ServiceAccount`, `Secret`,  or `ConfigMap` to create deployments, services, and to specify their configurations. 

* If you run {{ site.kic_product_name }} with database-backed {{site.base_gateway}}, you also need to have the permissions to create or update `PersistentVolumeClaim` to set volumes used for the database, and `Job` to run migration jobs. 

* If you want to enable auto scaling, you also need permissions to create or update `HorizontalPodAutoscaler`.
```

**Review by lena-larionova — CHANGES_REQUESTED** [review]

**Review by lena-larionova — APPROVED** [review]

**Pull request merged**